core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -1688,6 +1692,27 @@ static class GroupingImplementor implements AggImplementor {
     }
   }
 
+  /** Implementor for the {@code LITERAL_AGG} aggregate function. */
+  static class LiteralAggImplementor implements AggImplementor {
+    @Override public List<Type> getStateType(AggContext info) {
+      return ImmutableList.of();
+    }
+
+    @Override public void implementReset(AggContext info, AggResetContext reset) {
+    }
+
+    @Override public void implementAdd(AggContext info, AggAddContext add) {
+    }
+
+    @Override public Expression implementResult(AggContext info,
+        AggResultContext result) {
+      checkArgument(info.aggregation().kind == SqlKind.LITERAL_AGG);
+      checkArgument(result.call().rexList.size() == 1);
+      final RexNode rexNode = result.call().rexList.get(0);
```

Comment:
Should we check here that this rexNode is a RexLiteral?

Reply:
I don't feel strongly about it. There's some benefit to retaining flexibility.

For example, in `Sort` we didn't validate that `limit` and `offset` were literals and it turned out to be useful - because people wanted to use parameters.

Clearly the expression should not contain `RexInputRef`. But calls to parameter-less functions such as `CURRENT_DATE` seem like reasonable use cases.